### PR TITLE
Rename default_projector_$_id and use a list field for it.

### PR DESCRIPTION
**Pull request by reiterl**
Adds a new migration.

### CODE REVIEW

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    collection = "meeting"
+
+    def modify(self, object: Dict[str, Any]) -> None:
+        if "default_projector_$_id" in object:
+            object["default_projector_$_ids"] = object["default_projector_$_id"]
+            del object["default_projector_$_id"]
+        for name in cast(List[str], Meeting.default_projector__ids.replacement_enum):
+            old_field = f"default_projector_${name}_id"
+            new_field = f"default_projector_${name}_ids"
+            if old_field in object:
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:32]: Iterate over the object's fields instead and check if they start with `default_projector_$`. This way, you can remove the dependency to the `Meeting` object.

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                "default_projector_$_id",
+                *[
+                    f"default_projector_${name}_id"
+                    for name in cast(
+                        List[str], Meeting.default_projector__ids.replacement_enum
+                    )
+                ],
+            ):
+
+                if field in event.data:
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:58]: same here

File: openslides_backend/action/actions/meeting/replace_projector_id.py [hunk 3]
```diff
@@ -40,8 +40,11 @@ def get_updated_instances(self, payload: ActionData) -> ActionData:
             )
             changed = False
             for field in fields:
-                if meeting.get(field) == projector_id:
-                    instance[field] = meeting["reference_projector_id"]
+                change_list = meeting.get(field)
+                if change_list and projector_id in change_list:
+                    change_list.remove(projector_id)
+                    change_list.append(meeting["reference_projector_id"])
```
Comment (jsangmeister) [inline openslides_backend/action/actions/meeting/replace_projector_id.py:46]: only append the reference projector if the list is empty

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/action/actions/meeting/replace_projector_id.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/actions/meeting/replace_projector_id.py:46]: Done.

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py (hunk 1 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py]: Done.

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py]: Done.

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    """
+
+    target_migration_index = 35
+    collection = "meeting"
+
+    def modify(self, object: Dict[str, Any]) -> None:
+        for field in list(object.keys()):
+            if field == "default_projector_$_id":
+                object[field + "s"] = object[field]
+                del object["default_projector_$_id"]
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:27]:
```suggestion
                del object[field]
```

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            elif field.startswith("default_projector_$") and field.endswith("_id"):
+                old_field = field
+                new_field = field + "s"
+                if old_field in object:
+                    object[new_field] = [object[old_field]]
+                    del object[old_field]
```
Comment (jsangmeister) [inline openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:33]:
```suggestion
                object[field + "s"] = [object[field]]
                del object[field]
```
I would do it analogously to the first `if` and use the values directly. `old_field` is redundant anyway and doing the same thing twice in different ways is confusing.

Comment (jsangmeister) [reply openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:33]: Also, the last `if` is redundant

**Review by jsangmeister — CHANGES_REQUESTED** [review]

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:27]: Done.

File: openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py (hunk 5 above)
Comment (reiterl) [reply openslides_backend/migrations/migrations/0034_rename_change_default_projector_ids.py:33]: Done.

**Review by jsangmeister — APPROVED** [review]

**Comment by jsangmeister** [comment]
wait with merging for client and @emanuelschuetze's okay

**Pull request merged**